File: taska-backend/src/main/java/com/taska/dto/TaskRequest.java
package com.taska.dto;

import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record TaskRequest(
        @NotBlank String content,
        String description,
        UUID projectId,
        UUID sectionId,
        UUID parentId,
        Integer order,
        Integer priority,
        List<String> labels,
        LocalDate dueDate,
        LocalDateTime dueDateTime,
        Boolean isRecurring,
        Integer estimateMinutes,
        String mentionContext,
        String recurrenceRule
) {}

File: taska-android/app/src/main/java/com/taska/android/data/model/TaskRequest.kt
package com.taska.android.data.model

data class TaskRequest(
    val content: String,
    val type: String? = null,
    val description: String? = null,
    val projectId: String? = null,
    val parentId: String? = null,
    val priority: Int? = null,
    val labels: List<String>? = null,
    val scheduledAt: String? = null,
    val dueAt: String? = null,
    val allDay: Boolean? = null,
    val estimateMinutes: Int? = null,
    val isRecurring: Boolean? = null,
    val recurrenceRule: String? = null,
    val scope: String? = null,
    val occurrenceScheduledAt: String? = null
)
